dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/importers/ui/ImporterForm.java

```diff
@@ -0,0 +1,262 @@
+
+    private MetadataImporter importer;
+    private SafeBundleWrapper bundleWrapper;
+    private MetadataFormLookup lookup;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public ImporterForm() {
+        this.step = ImportStep.FIRST;
+    }
+
+
+    // -------------------- GETTERS --------------------
+
+    public List<FormItem> getItems() {
+        return items;
+    }
+
+    public List<ResultItem> getResultItems() {
+        return resultItems;
+    }
+
+    public ImportStep getStep() {
+        return step;
+    }
+
+    public ProcessingType[] getItemProcessingOptions(ResultItem item) {
+        return ItemType.VOCABULARY.equals(item.getItemType())
+                ? FormConstants.SINGLE_OPTIONS
+                : item.getMultipleAllowed()
+                    ? FormConstants.MULTIPLE_OPTIONS
+                    : FormConstants.SINGLE_OPTIONS;
+     }
+
+    // -------------------- LOGIC --------------------
+
+    public static ImporterForm createInitializedForm(MetadataImporter importer, Locale locale,
+                                                     Supplier<Map<MetadataBlock, List<DatasetFieldsByType>>> metadataSupplier) {
+        ImporterForm instance = new ImporterForm();
+        instance.initializeForm(importer, locale,
+                MetadataFormLookup.create(importer.getMetadataBlockName(), metadataSupplier));
+        return instance;
+    }
+
+    public void initializeForm(MetadataImporter importer, Locale locale, MetadataFormLookup lookup) {
+        this.lookup = lookup;
+        this.importer = importer;
+        this.bundleWrapper = new SafeBundleWrapper(importer, locale);
+        Tuple2<List<FormItem>, Map<ImporterFieldKey, FormItem>> itemsAndKeyToItem = initializeFormItems();
+        this.items = itemsAndKeyToItem._1;
+        this.keyToItem = itemsAndKeyToItem._2;
+    }
+
+    public void handleFileUpload(FileUploadEvent event) throws IOException {
+        FileUpload component = (FileUpload) event.getComponent();
+        removePreviousTempFile(component);
+
+        UploadedFile file = Optional.ofNullable(event)
+                .map(FileUploadEvent::getFile)
+                .orElseThrow(() -> new IllegalStateException("Null event or file"));
+        Path tempPath = prepareTempPath(file);
+        Files.copy(file.getInputStream(), tempPath, StandardCopyOption.REPLACE_EXISTING);
+        component.setValue(tempPath.toFile());
+    }
+
+    public void nextStep() {
+        Set<ValidationResult> validationResults = validateFormInputs();
+        if (!validationResults.isEmpty()) {
+            showValidationMessages(validationResults);
+            return;
+        }
+        List<ResultField> resultFields = importer.fetchMetadata(toImporterInput(items));
+        resultItems = new ResultItemsCreator(lookup).createItemsForView(resultFields);
+        step = ImportStep.SECOND;
+    }
+
+    public void onExit(Map<MetadataBlock, List<DatasetFieldsByType>> metadata) {
```

Comment:
Yes, but as this (and that from comment below) method is called directly from view, I'm not sure whether this is not too complicated name – If it would be called from backend only, then ok, but in this case I think that the meaning from the POV of user interaction flow is a bit more important than what it does internally.

Reply:
Makes sense, but isn't the method supposed to describe what is it doing and not when? When = "onExit".

Reply:
Ok, good point, you convinced me.